inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/HudiExtractNode.java

```diff
@@ -142,15 +143,17 @@ public Map<String, String> tableOptions() {
 
         // If the extend attributes starts with .ddl,
         // it will be passed to the ddl statement of the table
-        extList.forEach(ext -> {
-            String keyName = ext.get(EXTEND_ATTR_KEY_NAME);
-            if (StringUtils.isNoneBlank(keyName) &&
-                    keyName.startsWith(DDL_ATTR_PREFIX)) {
-                String ddlKeyName = keyName.substring(DDL_ATTR_PREFIX.length());
-                String ddlValue = ext.get(EXTEND_ATTR_VALUE_NAME);
-                options.put(ddlKeyName, ddlValue);
-            }
-        });
+        if (CollectionUtils.isNotEmpty(extList)) {
```

Comment:
I also think the same way. Initially, I prohibited 'primaryKey' from being empty, but I saw that there was already a 'PR' (#9323） solution to avoid setting 'primaryKey', so I changed it back. At the same time, the official website did not say that this configuration is mandatory. Finally, I believe it should not be empty, as normal use requires the use of primary keys. What do others think?

Reply:
1、the extList should add a check for npe, well done 
2、as for primary key, 
![image](https://github.com/apache/inlong/assets/26538404/1aed485f-b0d0-40e5-ab11-44cf962277c3)
 In hudi connector , the npe throws because we set string "null" for primary key but this "null" is not contained in hudi schema; that's the situation this pr #9323 trying to solve 
Actually hudi can run without pk (in this case the table won't support upsert) and the hoodie.datasource.write.recordkey.field is optional;

Reply:
see also in https://hudi.apache.org/docs/key_generation/

Reply:
OK